rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -705,10 +712,12 @@ void SqliteStorage::read_metadata()
         rcutils_time_point_value_t, std::string>();
 
       for (auto result : query_results) {
+        auto offered_qos_profiles = rosbag2_storage::to_rclcpp_qos_vector(
+          std::get<6>(result), metadata_.version);
```

Comment:
```suggestion
        auto offered_qos_profiles = rosbag2_storage::to_rclcpp_qos_vector(
        to offered_qos_profiles = rosbag2_storage::to_rclcpp_qos_vector(
          // Before db_schema_version_ = 3 we didn't store metadata in the database and real
          // metadata_.version will be lower than 9
          std::get<6>(result), (db_schema_version_ >= 3) ? metadata_.version : 8);
```

Reply:
Thanks for the background comment, I integrated this change :)